buildchain/buildchain/targets/local_image.py

```diff
@@ -62,10 +79,61 @@ def __init__(
             **kwargs
         )
 
-
     dockerfile   = property(operator.attrgetter('_dockerfile'))
     save_on_disk = property(operator.attrgetter('_save'))
     build_args   = property(operator.attrgetter('_build_args'))
+    dep_re = re.compile(
+        r'^\s*(COPY|ADD)( --[^ ]+)* (?P<src>[^ ]+) (?P<dst>[^ ]+)\s*$'
+    )
+
+    def load_deps_from_dockerfile(self) -> List[Tuple[int, str, str]]:
+        """Compute file dependencies from Dockerfile."""
+        dep_keywords = ('COPY', 'ADD')
+
+        with open(self.dockerfile, 'r') as dockerfile:
```

Comment:
Add `encoding='utf-8'`.

And IIRC `self.dockerfile` is a `Path` object so you can use `self.dockerfile.open('r', encoding='utf8')` or even `lines = self.dockerfile.read_text(encoding='utf-8').split('\n')`

Reply:
Stealing the first alternative, I find it more legible :)